tests/Baseline/hilti.optimization.const/noopt.hlt

```diff
@@ -6,43 +6,43 @@ import hilti;
 const bool t = True;
 const bool f = False;
 
-hilti::print(Foo::t, True);
-hilti::print(Foo::f, True);
+hilti::print(True, True);
+hilti::print(False, True);
 
-if ( t ) {
+if ( True ) {
     0;
 }
 else {
     1;
 }
 
 
-if ( f ) {
+if ( False ) {
     2;
 }
 else {
     3;
 }
 
 
-if ( t ) {
+if ( True ) {
     4;
 }
 
 
-if ( f ) {
+if ( False ) {
     5;
 }
```

Comment:
Just guessing, but you probably didn't move folding for `statement::If` into the new pass since we do it in the optimizer as well to remove dead code (which allows us to simplify parsers). This bothers me slightly more than not being able to handle feature constants there since even if you added it here, we also need that pass to happen in the optimizer. 😕 Not sure there's anything beyond explicitly calling the new folder again from the optimizer to get around that.

Reply:
I didn't. I think we'll need to revisit the relationship between the new folding and optimizer at some point, but I left that for later.